client/main/editors/transformeditor.js

```diff
@@ -0,0 +1,492 @@
+                let count = this._startsWithValidOps($formula);
+                if (count !== 0)
+                    $opEdit.css('width', (count+1) + 'ch');
+
+                if (dropdown.content() === this.opsToolbar)
+                    dropdown.hide();
+            }
+        });
+
+        if (hasOp) {
+            $opEdit = $('<div class="down-arrow">a</div>').appendTo($fp);
+            $opEdit.css('width', _sign.length + 'ch');
+            $opEdit.hide();
+
+            $opEdit.on('click', (event) => {
+                if (this._$wasEditingOpsFormula !== $formula || dropdown.content !== this.opsToolbar) {
+                    this.opsToolbar.show($formula);
+                    dropdown.show($formula, this.opsToolbar);
+
+                    let sel = window.getSelection();
+                    /*let range = sel.getRangeAt(0);
+                    let start = range.startOffset;
+                    let end = range.endOffset;
+                    let text = el.textContent;
+                    let before = text.substring(0, start);
+                    let after  = text.substring(end, text.length);*/
+
+                    let count = this._startsWithValidOps($formula);
+
+                    /*let textSelected = text.substring(start, end);
+                    el.textContent = (before + newText + after);*/
+                    sel.setBaseAndExtent($formula[0].firstChild, 0, $formula[0].firstChild, count);
+                    $formula[0].focus();
+                    //elements.$formula.focus();
+                    $opEdit.addClass('is-active');
+                }
+                event.stopPropagation();
+                event.preventDefault();
+            });
+
+            $opEdit.on('mousedown', (event) => {
+                this._$wasEditingOpsFormula = dropdown.focusedOn() !== null ? this.opsToolbar.focusedOn() : null;
+                elements._editorClicked = true;
+            });
+
+            $formula.on('editor:closing', () => {
+                $opEdit.removeClass('is-active');
+                elements.$showEditor.removeClass('is-active');
+            });
+        }
+
+        $formula.focus(() => {
+            keyboardJS.pause();
+        });
+        $formula.blur((event) => {
+            if (this._isRealBlur(elements)) {
+                elements._editorClicked = false;
+                return;
+            }
+
+            keyboardJS.resume();
+        });
+        $formula.on('keydown', (event) => {
+            if (event.keyCode === 13 && event.shiftKey === false) {    //enter
+                $formula.blur();
+                event.preventDefault();
+            }
+
+            if (event.keyCode === 9) {    //tab
+                event.preventDefault();
+            }
+        });
+
+        let $formulaMessageBox = $('<div class="formulaMessageBox""></div>').appendTo($formulaPair);
+        elements.$formulaMessage = $('<div class="formulaMessage""></div>').appendTo($formulaMessageBox);
```

Comment:
two many quotation marks? also, don't use camelCase in css classes

Reply:
good catch